### DISCUSSION TITLE
Hex polygon path misaligned

### DISCUSSION THREAD

**Discussion by JusticeMatthew** [matteobruni/tsparticles#4948] [Q&A]
Hello,

I'm using the hex path particles example found [here](https://codepen.io/matteobruni/pen/ZExaQyB)

Whenever I refresh the page a few of the particles are off the path that all the others are on causing the hex pattern to be messed up which you can see an example of here: 
![image](https://user-images.githubusercontent.com/72817096/232397932-57fc1814-b95e-444b-b71a-1cac4cb03890.png)

This ONLY happens when I refresh the page. If I trigger a reload of the page via some change in VS code then the path is drawn 100% perfectly 🤔 

I am using Next js and the project was created with create t3
Here is my particles component:
```javascript
import { useCallback, type FC } from "react";
import Particles from "react-tsparticles";
import { loadFull } from "tsparticles";
import { polygonPathName, loadPolygonPath } from "tsparticles-path-polygon";
import type { Engine } from "tsparticles-engine";

const HexParticles: FC = () => {
  const particlesInit = useCallback(async (engine: Engine) => {
    loadPolygonPath(engine);
    await loadFull(engine);
  }, []);

  return (
    <Particles
      init={particlesInit}
      className="absolute z-0 h-full w-full"
      options={{
        fullScreen: {
          enable: false,
        },
        particles: {
          color: {
            value: "#15db95",
          },
          move: {
            attract: {
              enable: false,
              distance: 10,
              rotate: {
                x: 2000,
                y: 2000,
              },
            },
            direction: "none",
            enable: true,
            outModes: {
              default: "destroy",
            },
            path: {
              clamp: false,
              enable: true,
              delay: {
                value: 0,
              },
              generator: polygonPathName,
              options: {
                sides: 6,
                turnSteps: 30,
                angle: 30,
              },
            },
            random: false,
            speed: 1,
            straight: false,
            trail: {
              fillColor: "#020202",
              length: 20,
              enable: true,
            },
          },
          number: {
            limit: 10,
            value: 0,
          },
          opacity: {
            value: 0.2,
          },
          shape: {
            type: "circle",
          },
          size: {
            value: 2,
          },
        },
        background: {
          color: "#000",
        },
        emitters: {
          autoPlay: true,
          direction: "none",
          life: {
            count: 1,
            duration: 10,
          },
          rate: {
            quantity: 1,
            delay: 1,
          },
          size: {
            width: 0,
            height: 0,
          },
          position: {
            x: 50,
            y: 50,
          },
        },
      }}
    />
  );
};

export default HexParticles;

```

You've been extremely helpful already so I'm hoping you can solve this one too! 💟

**Comment by matteobruni** [comment]
> Hello,
> 
> I'm using the hex path particles example found [here](https://codepen.io/matteobruni/pen/ZExaQyB)
> 
> Whenever I refresh the page a few of the particles are off the path that all the others are on causing the hex pattern to be messed up which you can see an example of here: ![image](https://user-images.githubusercontent.com/72817096/232397932-57fc1814-b95e-444b-b71a-1cac4cb03890.png)
> 
> This ONLY happens when I refresh the page. If I trigger a reload of the page via some change in VS code then the path is drawn 100% perfectly 🤔
> 
> I am using Next js and the project was created with create t3 Here is my particles component:
> 
> ```js
> import { useCallback, type FC } from "react";
> import Particles from "react-tsparticles";
> import { loadFull } from "tsparticles";
> import { polygonPathName, loadPolygonPath } from "tsparticles-path-polygon";
> import type { Engine } from "tsparticles-engine";
> 
> const HexParticles: FC = () => {
>   const particlesInit = useCallback(async (engine: Engine) => {
>     loadPolygonPath(engine);
>     await loadFull(engine);
>   }, []);
> 
>   return (
>     <Particles
>       init={particlesInit}
>       className="absolute z-0 h-full w-full"
>       options={{
>         fullScreen: {
>           enable: false,
>         },
>         particles: {
>           color: {
>             value: "#15db95",
>           },
>           move: {
>             attract: {
>               enable: false,
>               distance: 10,
>               rotate: {
>                 x: 2000,
>                 y: 2000,
>               },
>             },
>             direction: "none",
>             enable: true,
>             outModes: {
>               default: "destroy",
>             },
>             path: {
>               clamp: false,
>               enable: true,
>               delay: {
>                 value: 0,
>               },
>               generator: polygonPathName,
>               options: {
>                 sides: 6,
>                 turnSteps: 30,
>                 angle: 30,
>               },
>             },
>             random: false,
>             speed: 1,
>             straight: false,
>             trail: {
>               fillColor: "#020202",
>               length: 20,
>               enable: true,
>             },
>           },
>           number: {
>             limit: 10,
>             value: 0,
>           },
>           opacity: {
>             value: 0.2,
>           },
>           shape: {
>             type: "circle",
>           },
>           size: {
>             value: 2,
>           },
>         },
>         background: {
>           color: "#000",
>         },
>         emitters: {
>           autoPlay: true,
>           direction: "none",
>           life: {
>             count: 1,
>             duration: 10,
>           },
>           rate: {
>             quantity: 1,
>             delay: 1,
>           },
>           size: {
>             width: 0,
>             height: 0,
>           },
>           position: {
>             x: 50,
>             y: 50,
>           },
>         },
>       }}
>     />
>   );
> };
> 
> export default HexParticles;
> ```
> 
> You've been extremely helpful already so I'm hoping you can solve this one too! 💟

You're missing an `await` when calling `loadPolygonPath`, and it's better to call `loadFull` first

**Reply by JusticeMatthew** [reply]
Hmm tried that and still the same issue 😢 

If it helps, from what I've noticed it appears to be the very first particle spawned by the emitter is the one that's off the path

**Reply by matteobruni** [reply]
https://codesandbox.io/p/sandbox/throbbing-framework-cdz9lz

I've tested your component here, I've seen a small misalignment once in a while, and I think it's a decimal round issue, nothing to worry about. Can you see your problem there? I've noticed that the `loadPolygonPath` function was not awaitable, and this will be fixed in `2.10.0`

**Reply by JusticeMatthew** [reply]
Ah yes I should have mentioned that the `loadPolygonPath` was throwing a linting error when I tried to await it

It does also appear to happen with that codepen 😞 

It is a small but annoying issue however if it's something to do with the math rounding it doesn't sound like an "easy fix" for me so I'll probably just have to live with it haha
